src/scales/scale.time.js

```diff
@@ -50,246 +138,271 @@ module.exports = function(Chart) {
+				formattedTick = formatter(formattedTick, index, ticks);
 			}
 
-			var callback = helpers.valueOrDefault(tickOpts.callback, tickOpts.userCallback);
-
-			if (callback) {
-				return {
-					value: callback(formattedTick, index, ticks),
-					major: major
-				};
-			}
 			return {
 				value: formattedTick,
-				major: major
+				major: major,
+				time: time,
 			};
 		},
+
 		convertTicksToLabels: function() {
-			var me = this;
-			me.ticksAsTimestamps = me.ticks;
-			me.ticks = me.ticks.map(function(tick) {
-				return moment(tick);
-			}).map(me.tickFormatFunction, me);
-		},
-		getPixelForOffset: function(offset) {
-			var me = this;
-			var epochWidth = me.max - me.min;
-			var decimal = epochWidth ? (offset - me.min) / epochWidth : 0;
+			var ticks = this.ticks;
+			var i, ilen;
 
-			if (me.isHorizontal()) {
-				var valueOffset = (me.width * decimal);
-				return me.left + Math.round(valueOffset);
+			for (i=0, ilen=ticks.length; i<ilen; ++i) {
+				ticks[i] = this.tickFormatFunction(moment(ticks[i]));
 			}
+		},
 
-			var heightOffset = (me.height * decimal);
-			return me.top + Math.round(heightOffset);
+		/**
+		 * @private
+		 */
+		getPixelForOffset: function(time) {
+			var me = this;
+			var model = me._model;
+			var table = model.table;
+			var range = lookup(table, 'time', time);
+
+			// If value is out of bounds, use ticks [0, 1] or [n-1, n] for interpolation,
+			// note that the lookup table always contains at least 2 items (min and max)
+			var prev = !range.lo? table[0] : !range.hi? table[table.length-2] : range.lo;
+			var next = !range.lo? table[1] : !range.hi? table[table.length-1] : range.hi;
+
+			var span = next.time - prev.time;
+			var ratio = span? (time - prev.time) / span : 0;
+			var offset = (next.decimal - prev.decimal) * ratio;
+			var size = model.horizontal? me.width : me.height;
+			var start = model.horizontal? me.left : me.top;
+
+			return start + size * (prev.decimal + offset);
 		},
+
 		getPixelForValue: function(value, index, datasetIndex) {
 			var me = this;
-			var offset = null;
+			var time = null;
+
 			if (index !== undefined && datasetIndex !== undefined) {
-				offset = me._parsedData.datasets[datasetIndex][index];
+				time = me._model.datasets[datasetIndex][index];
 			}
 
-			if (offset === null) {
-				if (!value || !value.isValid) {
-					// not already a moment object
-					value = timeHelpers.parseTime(me, me.getRightValue(value));
-				}
-
-				if (value && value.isValid && value.isValid()) {
-					offset = value.valueOf();
-				}
+			if (time === null) {
+				time = parse(value, me);
 			}
 
-			if (offset !== null) {
-				return me.getPixelForOffset(offset);
+			if (time !== null) {
+				return me.getPixelForOffset(time);
 			}
 		},
+
 		getPixelForTick: function(index) {
-			return this.getPixelForOffset(this.ticksAsTimestamps[index]);
+			return index >= 0 && index < this.ticks.length?
+				this.getPixelForOffset(this.ticks[index].time) :
+				null;
```

Comment:
We didn't commit on that rule and ESLint can be tweaked to disable it. I tried to get used to it but I still prefer that form since it mimics the if/else indentation. Having `null` on the previous line doesn't make it more readable: should be 1 line or 3, not 2. And since having all in one line is too long, I definitely prefer this layout. That's really minor though.